optaplanner-core/src/test/java/org/optaplanner/core/impl/solver/DefaultSolverTest.java

```diff
@@ -77,12 +82,122 @@ public void solveLegacy() {
         assertSame(solution, solver.getBestSolution());
     }
 
+    @Test
+    public void solveEmptyEntityList() {
+        SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class)
+                .withPhases(new CustomPhaseConfig().withCustomPhaseCommands(
+                        scoreDirector -> fail("All phases should be skipped because there are no movable entities.")
+                ));
+        SolverFactory<TestdataSolution> solverFactory = SolverFactory.create(solverConfig);
+        Solver<TestdataSolution> solver = solverFactory.buildSolver();
+
+        TestdataSolution solution = new TestdataSolution("s1");
+        solution.setValueList(Arrays.asList(new TestdataValue("v1"), new TestdataValue("v2")));
+        solution.setEntityList(Collections.emptyList());
+
+        solution = solver.solve(solution);
+        assertNotNull(solution);
+        assertEquals(true, solution.getScore().isSolutionInitialized());
+        assertSame(solution, solver.getBestSolution());
+    }
+
+    @Test
+    public void solveChainedEmptyEntityList() {
+        SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataChainedSolution.class, TestdataChainedEntity.class)
+                .withPhases(new CustomPhaseConfig().withCustomPhaseCommands(
+                        scoreDirector -> fail("All phases should be skipped because there are no movable entities.")
+                ));
+        SolverFactory<TestdataChainedSolution> solverFactory = SolverFactory.create(solverConfig);
+        Solver<TestdataChainedSolution> solver = solverFactory.buildSolver();
+
+        TestdataChainedSolution solution = new TestdataChainedSolution("s1");
+        solution.setChainedAnchorList(Arrays.asList(new TestdataChainedAnchor("v1"), new TestdataChainedAnchor("v2")));
+        solution.setChainedEntityList(Collections.emptyList());
+
+        solution = solver.solve(solution);
+        assertNotNull(solution);
+        assertEquals(true, solution.getScore().isSolutionInitialized());
+        assertSame(solution, solver.getBestSolution());
+    }
+
+    @Test @Ignore("We currently don't support an empty value list yet if the entity list is not empty.")
```

Comment:
Is there a follow-up Jira ticket for this? If so, let's link it here, please.

Reply:
Done, added https://issues.jboss.org/browse/PLANNER-1738